config_builder.py

```diff
@@ -0,0 +1,251 @@
+#!/opt/datadog-agent/embedded/bin/python
+'''
+This script is used to generate the configuration of the datadog agent, its
+integrations and other moving parts.
+'''
+
+from os import getenv, environ
+from os.path import isfile
+import logging
+from urllib2 import urlopen, URLError, HTTPError
+from socket import getdefaulttimeout, setdefaulttimeout
+from ConfigParser import ConfigParser
+import yaml
+
+class ConfBuilder(object):
+    '''
+    This class manages the configuration files
+    '''
+    def __init__(self):
+        # excludes from the generic variables parsing the ones that have a
+        # certain logic warpped around them
+        self.exclude_from_generic = [
+            'DD_API_KEY', 'DD_API_KEY_FILE', 'DD_HOME',
+            'DD_START_AGENT', 'DD_LOGS_STDOUT'
+            ]
+        dd_agent_root = '/etc/dd-agent'
+        dd_home = getenv('DD_HOME')
+        if dd_home is not None:
+            dd_agent_root = '{}/agent'.format(dd_home)
+        self.datadog_conf_file = '{}/datadog.conf'.format(dd_agent_root)
+        self.supervisor_conf_file = '{}/supervisor.conf'.format(dd_agent_root)
+        self.conf_d = '{}/conf.d'.format(dd_agent_root)
+        # This will store the config parser object that is used in the different functions
+        self.config = None
+
+    def set_instance_property(self, integration, property_key, property_value):
+        '''
+        Sets a property in the instances of an integration file
+        '''
+        _source = '{}/{}.yaml.example'.format(self.conf_d, integration)
+        _target = '{}/{}.yaml'.format(self.conf_d, integration)
+        _data = {}
+        # Do not overwrite target if exists
+        if isfile(_target):
+            _source = _target
+        with open(_source, 'r') as _stream:
+            _data = yaml.load(_stream)
+
+        # Hack to not end up with 'null' as init_config value
+        if _data.get('init_config') is None:
+            _data['init_config'] = {}
+
+        _instances = _data.get('instances')
+        for _inst in _instances:
+            _inst[property_key] = property_value
+
+        with open(_target, 'w') as _stream:
+            yaml.dump(_data, _stream, default_flow_style=False)
+
+    def build_integration_conf(self):
```

Comment:
> I was thinking about adding the ability to have properties via a hash, not only 1 property at a time

dict or individual, this is not an API we want to support. Integration configs should be all or nothing. Those we touch in entrypoints are special cases that we'll work on getting rid of, not extending.